src/provider/provider_tracking.c

```diff
@@ -343,6 +348,15 @@ static umf_result_t trackingFree(void *hProvider, void *ptr, size_t size) {
         umf_ba_global_free(value);
     }
 
+    // umfMemoryProviderFree() should not be called with size == 0,
```

Comment:
But according to this PR  `umfMemoryProviderFree()` for tracking provider can be called with `size == 0`.

Maybe a particular pool implementation should explicitly get the size from the tracker and call `umfMemoryProviderFree()` with the right size?

Reply:
Right. New fix: https://github.com/oneapi-src/unified-memory-framework/pull/487